src/ResourceAdequacy/results/debug.jl

```diff
@@ -0,0 +1,142 @@
+struct Debug <: ResultSpec end
+
+struct DebugResult{
+    N, # Number of timesteps simulated
+    L, # Length of each timestep
+    T <: Period, # Units of timestep duration
+    P <: PowerUnit, # Units for power results
+    E <: EnergyUnit, # Units for energy results
+    SS <: SimulationSpec
+} <: Result{N,L,T,SS}
+
+    regions::Vector{String}
+    interfaces::Vector{Tuple{Int,Int}}
+    timestamps::StepRange{ZonedDateTime,T}
+
+    lole::LOLE{N,L,T}
+    regionloles::Vector{LOLE{N,L,T}}
+    periodlolps::Vector{LOLP{L,T}}
+    regionalperiodlolps::Matrix{LOLP{L,T}}
+
+    eue::EUE{N,L,T,E}
+    regioneues::Vector{EUE{N,L,T,E}}
+    periodeues::Vector{EUE{1,L,T,E}}
+    regionalperiodeues::Matrix{EUE{1,L,T,E}}
+
+    flows::Matrix{ExpectedInterfaceFlow{1,L,T,P}}
+    utilizations::Matrix{ExpectedInterfaceUtilization{1,L,T}}
+
+    gens_available::Array{Bool,3}
+    lines_available::Array{Bool,3}
+    stors_available::Array{Bool,3}
+    genstors_available::Array{Bool,3}
+
+    sample_ues::Vector{Float64}
+
+    simulationspec::SS
+
+    function DebugResult{}(
+        regions::Vector{String}, interfaces::Vector{Tuple{Int,Int}},
+        timestamps::StepRange{ZonedDateTime,T},
+        lole::LOLE{N,L,T}, regionloles::Vector{LOLE{N,L,T}},
+        periodlolps::Vector{LOLP{L,T}},
+        regionalperiodlolps::Matrix{LOLP{L,T}},
+        eue::EUE{N,L,T,E}, regioneues::Vector{EUE{N,L,T,E}},
+        periodeues::Vector{EUE{1,L,T,E}},
+        regionalperiodeues::Matrix{EUE{1,L,T,E}},
+        flows::Matrix{ExpectedInterfaceFlow{1,L,T,P}},
+        utilizations::Matrix{ExpectedInterfaceUtilization{1,L,T}},
+        gens_available::Array{Bool,3}, lines_available::Array{Bool,3},
+        stors_available::Array{Bool,3}, genstors_available::Array{Bool,3},
+        sample_ues::Vector{Float64}, simulationspec::SS
+) where {N,L,T,P,E,SS}
+
+        nregions = length(regions)
+        ninterfaces = length(interfaces)
+        ntimesteps = length(timestamps)
+
+        @assert ntimesteps == N
+
+        @assert length(regionloles) == nregions
+        @assert length(periodlolps) == ntimesteps
+        @assert size(regionalperiodlolps) == (nregions, ntimesteps)
+
+        @assert length(regioneues) == nregions
+        @assert length(periodeues) == ntimesteps
+        @assert size(regionalperiodeues) == (nregions, ntimesteps)
+
+        @assert size(flows) == (ninterfaces, ntimesteps)
+        @assert size(utilizations) == (ninterfaces, ntimesteps)
```

Comment:
It would be good to check that `gens_available` and friends are the right size here - same for `sample_ues`